beamformer.c

```diff
@@ -73,17 +76,17 @@ alloc_shader_storage(BeamformerCtx *ctx, Arena a)
 		break;
 	case GL_VENDOR_NVIDIA:
 		cs->raw_data_arena = os_alloc_arena(cs->raw_data_arena, full_rf_buf_size);
-		if (g_cuda_lib_functions[CLF_REGISTER_BUFFER]) {
-			cuda_register_buffer *fn = g_cuda_lib_functions[CLF_REGISTER_BUFFER];
-			/* TODO: this needs the correct parameters */
-			fn(NULL, cs->raw_data_ssbo);
+		if (g_cuda_lib_functions[CLF_REGISTER_BUFFERS]) {
+			register_cuda_buffers *fn = g_cuda_lib_functions[CLF_REGISTER_BUFFERS];
+			fn(cs->rf_data_ssbos, ARRAY_COUNT(cs->rf_data_ssbos), cs->raw_data_ssbo);
+		}
+		if (g_cuda_lib_functions[CLF_INIT_CONFIG]) {
+			init_cuda_configuration* fn = g_cuda_lib_functions[CLF_INIT_CONFIG];
+			fn((u32*)&(bp->rf_raw_dim), (u32*)&(bp->dec_data_dim), bp->channel_mapping, bp->channel_offset > 0);
```

Comment:
Are these both required? If so they should be behind a single if.

> `(u32*)&(bp->rf_raw_dim)`

`bp->rf_raw_dim.E` etc.

Reply:
Updated